src/index.js

```diff
@@ -1,13 +1,141 @@
-    return "결과 값 String";
-  };
-}
+  const answer = [];
+
+  const userInput = document.querySelector('#user-input');
+  const submitButton = document.querySelector('#submit');
+  const resultDiv = document.querySelector('#result');
+
+  function getRandomDecimalLessThan(num) {
+    return Math.floor((Math.random() * 10) % num);
+  }
+
+  function setRandomAnswer() {
+    const array = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+    answer.length = 0;
+
+    for (let i = 0; i < 3; i += 1) {
+      const idx = getRandomDecimalLessThan(array.length);
+      answer.push(array[idx]);
+      array.splice(idx, 1);
+    }
+    return answer;
+  }
+
+  function initAnswer() {
+    answer.length = 0;
+  }
+
+  function resetResultStr() {
+    resultDiv.innerHTML = '';
+  }
+
+  function resetUserInput() {
+    userInput.value = '';
+  }
+
+  function resetGame() {
+    resetResultStr();
+    resetUserInput();
+    initAnswer();
+    setRandomAnswer();
+  }
+
+  function handleError() {
+    alert(`잘못된 값을 입력하셨습니다.
+1부터 9까지의 서로 다른 수로 이루어진 3자리 숫자를 입력해주세요.
+예시) 123, 645, 987`);
+    return false;
+  }
+
+  function isValidInput(input) {
+    if (Number.isNaN(input) || input < 123 || input > 987) {
+      return false;
+    }
+    const set = new Set();
+    let num = input;
 
-// export default class BaseballGame {
-//   play(computerInputNumbers, userInputNumbers) {
-//     return "결과 값 String";
-//   }
-// }
+    for (let i = 0; i < 3; i += 1) {
+      set.add(num % 10);
+      num = Math.floor(num / 10);
+    }
+
+    if (set.size !== 3 || set.has(0)) {
+      return false;
+    }
+    return true;
+  }
+
+  function makeGameResult(input) {
+    const inputArray = [];
+    const answerArray = answer.slice();
+    const obj = {
+      balls: 0,
+      strikes: 0,
+    };
+    let num = input;
+
+    for (let i = 0; i < 3; i += 1) {
+      inputArray.splice(0, 0, num % 10);
+      num = Math.floor(num / 10);
+    }
+
+    for (let i = 0; i < inputArray.length; i += 1) {
+      if (inputArray[i] === answerArray[i]) {
+        obj.strikes += 1;
+        inputArray.splice(i, 1);
+        answerArray.splice(i, 1);
+        i -= 1;
+      }
+    }
+    for (let i = 0; i < inputArray.length; i += 1) {
+      obj.balls += (answerArray.includes(inputArray[i]) ? 1 : 0);
+    }
+    return obj;
+  }
+
+  function play(computerInputNumbers, userInputNumbers) {
+    if (!isValidInput(userInputNumbers)) {
+      return handleError();
+    }
+    const gameResult = makeGameResult(userInputNumbers);
+    let resultStr = '';
+    if (gameResult.balls > 0) {
+      resultStr += `${gameResult.balls}볼`;
+    }
+    if (gameResult.balls > 0 && gameResult.strikes > 0) {
+      resultStr += ' ';
+    }
+    if (gameResult.strikes > 0) {
+      resultStr += `${gameResult.strikes}스트라이크`;
+    }
+    return (resultStr.length === 0 ? '낫싱' : resultStr);
+  }
+
+  function handleSubmit() {
+    const result = play(answer, +(userInput.value));
```

Comment:
코드를 살펴보니 answer는 배열이고 +(userInput.value)는 숫자로 처리되는 것 같습니다. 물론 저희는 앞으로 TS를 사용하겠지만, 타이핑이 없는 JS에서는 최대한 타입에 주의해야됩니다.

현재 play 함수의 매개변수의 이름을 보면 두 변수가 타입이 같다고 읽히게 됩니다. 이는 이후 다른 개발자와 협업할 때 타입 실수가 자주 일어나는 원인이 될 것 같습니다.

제 개인적인 추천으로는, 변수명이 비슷함으로 최대한 타입도 같게 처리를 하면 좀 더 일기 편한 코드가 되지 않았을까 생각이 듭니다.

Reply:
```suggestion
    const result = play(answer, Number(userInput.value));
```
또한 더욱 명시적으로 형변환을 작성해주는게 더 좋을것 같습니다.

Reply:
매개변수의 자료형이 다른 것이 확실히 이상하게 느껴지네요... 함수의 처리 방식을 고쳐보겠습니다. 형변환 조언도 감사합니다!